atlas_densities/densities/mtype_densities_from_map/utils.py

```diff
@@ -58,3 +56,18 @@ def _check_probability_map_consistency(
             "The following molecular types are missing in the probability map: %s",
             molecular_types_diff,
         )
+
+
+def _merge_probability_maps(probability_maps: List["pd.Dataframe"]) -> "pd.Dataframe":
```

Comment:
@GianlucaFicarelli is there a more pandas-y way of doing this?

Reply:
As a clarification, are both of these cases considered a conflict?
1. two probability maps contain the same region and the same molecular_type
2. two probability maps contain the same region but different molecular_type

-- 1.
If only the case 1 is a conflict, and if we suppose that having dataframes with overlapping indices is not the common case, then a possibility is to optimistically concat the dataframes as it's already done, but using `pd.concat(..., verify_integrity=True)`, unless it's more expensive (to be verified)..

Alternatively, it can be checked manually with:
```python
result.index.duplicated().any()
```

-- 2.
Instead, if also the case 2 is a conflict, it could be checked with:
```python
arr = np.concatenate([df.index.get_level_values("region").drop_duplicates() for df in probability_maps])
if len(arr) != len(np.unique(arr)): ...
```

I'm not sure which approach is more performant, it could depend on the size, number, and content of the dataframes.